Rework paradrop logic to be more robust.

pchote opened the pull request:
The current paradrop logic works by having the aircraft blindly insert the actor into the world, relying on the `Parachute` activity to set the cell and position.  The actor doesn't have a previously set position, so appears for one tick (before `Parachute` runs) at 0,0. I don't have a repro case for #9750/#15520 but i'm reasonably confident that these happen because something cancels `Parachute` before it has a chance to tick.

This PR fixes the mess by making the aircraft insert the unit directly into the desired cell, and now correctly accounts for sub-cell positions. Multiple units can now be dropped in the same cell, with a new `DropInterval` parameter controlling how spaced out they should be.

Fixes #3409
Fixes #9750
Fixes #15520

reaperrr commented on OpenRA.Mods.Common/Traits/ParaDrop.cs:
```diff
@@ -70,30 +83,38 @@ void ITick.Tick(Actor self)
+			var dropActor = cargo.Peek(self);
+			var dropPositionable = dropActor.TraitOrDefault<IPositionable>();
+			var dropCell = self.Location;
+			var dropSubCell = dropPositionable.GetAvailableSubCell(dropCell);
```
Won't this crash if `dropPositionable` is null? In which case you might as well drop the `OrDefault` above.

reaperrr reviewed (commented):
Infantry spacing of RA paradrop looks a bit weird now

reaperrr commented:
Also needs a rebase.

pchote commented:
Fixed and rebased.

> Infantry spacing of RA paradrop looks a bit weird now

This is a consequence of using subcells for positions, instead of the middle of each cell. I'm not sure there is much we can do about this.

matjaeck commented:
> Infantry spacing of RA paradrop looks a bit weird now.

Not a problem IMO.

PunkPun reviewed (commented):
This doesn't fully fix the corner bug. On bleed & release if "status bar" option is set to "always on" you are able to see health bars of soldiers in the corner. In this pr it is not the case, but infantry still provide vision for a tick in the corner

matjaeck commented:
> infantry still provide vision for a tick in the corner

I think we can live with this.

pchote commented:
> In this pr it is not the case, but infantry still provide vision for a tick in the corner

I can't reproduce this on this pr. Can anyone else?

pchote commented:
Fixed.

Punsho commented:
I often get that ra paradrop doesn't drop all 5 soldiers in a run if rubble is close. It's way worse then in the current release

pchote committed:
Rework paradrop logic to be more robust.

pchote commented:
I've halved the inter-drop delay, which should help with the spacing and missing soldier issues.

reaperrr commented:
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4a1b9262b78864ce5f057da5290e8d6aab4a116f...f473e8d9bfafbac6b9a4f2b95e10a8860ffefed3
